ballerina/tests/fromXml_test.bal

```diff
@@ -1620,6 +1620,374 @@ function testCommentMiddleInContent2() returns error? {
+    |} rec = check fromXmlStringWithType(xmlStr);
+    test:assertEquals(rec.length(), 1);
+    test:assertEquals(rec.Employee, [
+        {
+            "Name": "John Doe",
+            "Age": "30"
+        },
+        {
+            "Name": "Kanth Kevin",
+            "Age": "26"
+        }
+    ]);
+
+    string xmlStr2 = string `<Org>
+                        <name>WSO2</name>
+                        <name>Apple</name>
+                    </Org>`;
+    record {
+        json[] name;
+    } rec2 = check fromXmlStringWithType(xmlStr2);
+    test:assertEquals(rec2.length(), 1);
+    test:assertEquals(rec2.name, ["WSO2", "Apple"]);
+}
+
+@test:Config
+function testAnydataArrayAsFieldTypeWiThFromXmlWithType() returns error? {
+    xml xmlVal = xml `<Company>
+                        <Employee>
+                            <Name>John Doe</Name>
+                            <Age>30</Age>
+                        </Employee>
+                        <Employee>
+                            <Name>Kanth Kevin</Name>
+                            <Age>26</Age>
+                        </Employee>
+                    </Company>`;
+    record {|
+        anydata[] Employee;
+    |} rec = check fromXmlWithType(xmlVal);
+    test:assertEquals(rec.length(), 1);
+    test:assertEquals(rec.Employee, [
+        {
+            "Name": "John Doe",
+            "Age": "30"
+        },
+        {
+            "Name": "Kanth Kevin",
+            "Age": "26"
+        }
+    ]);
+
+    xml xmlVal2 = xml `<Company>
+                        <name>WSO2</name>
+                        <name>Apple</name>
+                    </Company>`;
+    record {
+        anydata[] name;
+    } rec2 = check fromXmlWithType(xmlVal2);
+    test:assertEquals(rec2.length(), 1);
+    test:assertEquals(rec2.name, ["WSO2", "Apple"]);
+}
+
+@test:Config
+function testJsonArrayAsFieldTypeWiThFromXmlWithType() returns error? {
+    xml xmlVal = xml `<Company>
+                        <Employee>
+                            <Name>John Doe</Name>
+                            <Age>30</Age>
+                        </Employee>
+                        <Employee>
+                            <Name>Kanth Kevin</Name>
+                            <Age>26</Age>
+                        </Employee>
+                    </Company>`;
+    record {|
+        json[] Employee;
+    |} rec = check fromXmlWithType(xmlVal);
+    test:assertEquals(rec.length(), 1);
+    test:assertEquals(rec.Employee, [
+        {
+            "Name": "John Doe",
+            "Age": "30"
+        },
+        {
+            "Name": "Kanth Kevin",
+            "Age": "26"
+        }
+    ]);
+
+    xml xmlVal2 = xml `<Org>
+                        <name>WSO2</name>
+                        <name>Apple</name>
+                    </Org>`;
+    record {
+        json[] name;
+    } rec2 = check fromXmlWithType(xmlVal2);
+    test:assertEquals(rec2.length(), 1);
+    test:assertEquals(rec2.name, ["WSO2", "Apple"]);
+}
```

Comment:
Shall we add tests to check the behaviour of `sequesnces, text, instructions`.

Reply:
This test is only added to validate the Fix of this issue https://github.com/ballerina-platform/ballerina-library/issues/6092. Mentioned tests are already available